tools/src/main/scala/scala/scalanative/codegen/Lower.scala

```diff
@@ -1260,14 +1283,31 @@ object Lower {
         Sig.Method("alloc", Seq(Type.Int, arrcls))
       )
   }.toMap
-  val arrayAllocSig = Type.typeToArray.map {
+  val arrayAllocInHeapSig = Type.typeToArray.map {
     case (ty, arrname) =>
       val Global.Top(id) = arrname: @unchecked
       ty -> Type.Function(
         Seq(Type.Ref(Global.Top(id + "$")), Type.Int),
         Type.Ref(arrname)
       )
   }.toMap
+  val arrayAllocInZone = Type.typeToArray.map {
+    case (ty, arrname) =>
+      val Global.Top(id) = arrname: @unchecked
+      val arrcls = Type.Ref(arrname)
+      ty -> Global.Member(
+        Global.Top(id + "$"),
+        Sig.Method("alloc", Seq(Type.Int, Type.Ptr, arrcls))
+      )
+  }.toMap
+  val arrayAllocInZoneSig = Type.typeToArray.map {
+    case (ty, arrname) =>
+      val Global.Top(id) = arrname: @unchecked
+      ty -> Type.Function(
+        Seq(Type.Ref(Global.Top(id + "$")), Type.Int, Type.Ptr),
+        Type.Ref(arrname)
+      )
+  }.toMap
```

Comment:
These definitions are duplicating the `arrayAllocInHeap` and `arrayAllocInHeapSig` with just different signatures. We can probably refactor it a bit, to remove the duplicates, eg. using a common function